src/workerd/server/server.c++

```diff
@@ -1303,10 +1304,79 @@ private:
   capnp::Capability::Client getCapability(uint channel) override {
     KJ_FAIL_REQUIRE("no capability channels");
   }
+  class CacheClientImpl final: public CacheClient {
+  public:
+    CacheClientImpl(Service& parent)
+        : parent(parent) {
+    }
+
+    kj::Own<kj::HttpClient> getDefault(kj::Maybe<kj::String> cfBlobJson,
+                                       kj::Maybe<Tracer::Span&> parentSpan) override {
+
+      return kj::heap<CacheHttpClientImpl>(
+          parent, nullptr, kj::mv(cfBlobJson), kj::mv(parentSpan));
+    }
+
+    kj::Own<kj::HttpClient> getNamespace(
+        kj::StringPtr cacheName, kj::Maybe<kj::String> cfBlobJson,
+        kj::Maybe<Tracer::Span&> parentSpan) override {
+
+      auto encodedName = kj::encodeUriComponent(cacheName);
```

Comment:
Can you move this encoding step to happen inside addCacheNameHeader? So it's with the other code related to how this is used.